src/components/charts/RavijenChart.tsx

```diff
@@ -11,7 +11,7 @@ import CategoryChart from './ravijenChart/CategoryChart'
 
 type Props = {
   categories: DottedName[]
-  subcategories: Record<DottedName, DottedName[]>
+  subcategories: Record<DottedName, DottedName[]> | undefined
```

Comment:
```suggestion
  subcategories?: Record<DottedName, DottedName[]> | undefined
```

Mark it optional then (but keep undefined)

That's a good practice because when you want create the props you will have to specify `undefined`

Example

```typescript
type MyType = {
  myProp: string | undefined
}

const toto: MyType = {
  myProp: undefined // mandatory here
}

type MyTypeWithOptional = {
  myProp?: string | undefined
}

const tata: MyTypeWithOptional = {} // no more

type MyTypeWithOptionalAndNoUndefined = {
  myProp?: string
}

const tutu: MyTypeWithOptionalAndNoUndefined = {
  myProp: undefined
} // works but not really perfect
````

Reply:
Ahh okay good to know